Bots/DotNet/Consumers/CodeFirst/WaterfallHostBot/TokenExchangeSkillHandler.cs

```diff
@@ -29,39 +29,37 @@ public class TokenExchangeSkillHandler : SkillHandler
 
         private readonly BotAdapter _adapter;
         private readonly SkillsConfiguration _skillsConfig;
-        private readonly SkillHttpClient _skillClient;
         private readonly string _botId;
         private readonly SkillConversationIdFactoryBase _conversationIdFactory;
         private readonly ILogger _logger;
-        private readonly IExtendedUserTokenProvider _tokenExchangeProvider;
-        private readonly IConfiguration _configuration;
+        private readonly string _connectionName;
+        private readonly AuthenticationConfiguration _authConfig;
+        private readonly BotFrameworkAuthentication _botAuth;
 
         public TokenExchangeSkillHandler(
             BotAdapter adapter,
             IBot bot,
             IConfiguration configuration,
-            SkillConversationIdFactoryBase conversationIdFactory,
-            SkillsConfiguration skillsConfig,
-            SkillHttpClient skillClient,
             ICredentialProvider credentialProvider,
+            SkillConversationIdFactoryBase conversationIdFactory,
             AuthenticationConfiguration authConfig,
+            BotFrameworkAuthentication botAuth,
+            SkillsConfiguration skillsConfig,
             IChannelProvider channelProvider = null,
             ILogger<TokenExchangeSkillHandler> logger = null)
             : base(adapter, bot, conversationIdFactory, credentialProvider, authConfig, channelProvider, logger)
         {
             _adapter = adapter;
-            _tokenExchangeProvider = adapter as IExtendedUserTokenProvider;
-            if (_tokenExchangeProvider == null)
-            {
-                throw new ArgumentException($"{nameof(adapter)} does not support token exchange");
-            }
 
-            _configuration = configuration;
-            _skillsConfig = skillsConfig;
-            _skillClient = skillClient;
+            _botAuth = botAuth;
+            _authConfig = authConfig;
             _conversationIdFactory = conversationIdFactory;
-            _logger = logger ?? NullLogger<TokenExchangeSkillHandler>.Instance;
+            _skillsConfig = skillsConfig ?? new SkillsConfiguration(configuration);
             _botId = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppIdKey)?.Value;
+            _logger = logger;
+
+            var settings = configuration.GetSection("Bot.Builder.Community.Components.TokenExchangeSkillHandler")?.Get<ComponentSettings>() ?? new ComponentSettings();
+            _connectionName = settings.TokenExchangeConnectionName ?? configuration.GetSection("SsoConnectionName")?.Value;
```

Comment:
Should we update AppSettings.json to include this new section?

Reply:
I'll remove it and stay with how we used to get connection name before.